lib/gman_client/api/health_check.rb

```diff
@@ -0,0 +1,16 @@
+module GmanClient
+  module Api
+    # HealthCheck API
+    module HealthCheck
+      # Retrieve a health check
+      def health_check
+        response = request.api.v1.health_check.get
+
+        unless response.to_h[:health_check] == 'Passed'
```

Comment:
I think it would make more sense for the agent to raise the exception as I feel the client has done its job in regards to the health check (is it healthy, yes, no?)... and its the consumers responsibility to decide what to do with that.